packages/components/src/__future__/Tabs/_docs/Tabs.spec.stories.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import { Meta, StoryObj } from '@storybook/react'
+import { within, userEvent, expect } from '@storybook/test'
+import { Text } from '~components/Text'
+import { Tab, TabList, TabPanel, Tabs } from '../index'
+
+const meta = {
+  title: 'Components/Tabs/Tabs (Future)/Tests',
+  parameters: {
+    controls: { disable: true },
+  },
+} satisfies Meta
+
+export default meta
+
+type Story = StoryObj<typeof meta>
+
+export const ArrowsShowingAndHiding: Story = {
+  render: () => {
+    return (
+      <div style={{ maxWidth: '500px' }}>
+        <Tabs>
+          <TabList aria-label="Tabs">
+            <Tab id="one">Tab 1</Tab>
+            <Tab id="two">Tab 2</Tab>
+            <Tab id="three" badge="3">
+              Tab 3
+            </Tab>
+            <Tab id="disabled" isDisabled>
+              Disabled Tab
+            </Tab>
+            <Tab id="four">Tab 4</Tab>
+            <Tab id="five">Tab 5</Tab>
+          </TabList>
+          <TabPanel id="one" className="p-24">
+            <Text variant="body">Content 1</Text>
+          </TabPanel>
+          <TabPanel id="two" className="p-24">
+            <Text variant="body">Content 2</Text>
+          </TabPanel>
+          <TabPanel id="three" className="p-24">
+            <Text variant="body">Content 3</Text>
+          </TabPanel>
+          <TabPanel id="disabled" className="p-24">
+            <Text variant="body">Disabled content</Text>
+          </TabPanel>
+          <TabPanel id="four" className="p-24">
+            <Text variant="body">Content 4</Text>
+          </TabPanel>
+          <TabPanel id="five" className="p-24">
+            <Text variant="body">Content 5</Text>
+          </TabPanel>
+        </Tabs>
+      </div>
+    )
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement.parentElement!)
+
+    expect(canvas.queryByTestId('kz-tablist-left-arrow')).not.toBeInTheDocument()
+
+    const rightArrow = await canvas.findByTestId('kz-tablist-right-arrow')
+
+    await userEvent.click(rightArrow)
+    await new Promise((r) => setTimeout(r, 500))
```

Comment:
`await waitFor` the left arrow should hopefully allow you to not need a timeout :)

Reply:
For the first one yes, but then I'll still need the timeout when there's no arrow appearance/disappearance to hook into. So is it worth doing it, or just keep it consistent with the timeout

Reply:
Checking that the element is not in the dom when you expect it to disappear probably should work?
I'd just try to avoid timeouts when you have an explicit result you're waiting for, because it's still potentially flaky. It only makes sense to use it when the thing you're testing has functionality that relies on timeouts. It's okay as a last resort if nothing else works though.

Reply:
The timeout is actually more robust than that. Your idea works okay when it's testing for disappearance, but hooking into appearance actually breaks this. 

The problem with testing for appearance/disappearance is that that doesn't mean the animation is done. The sequence goes:
  * Press right -> animation starts -> left arrow appears -> animation finishes. 

Even with disappearance, it's closer, but it still doesn't mean the animation is done because of the threshold of 75% (it disappears before the element is completely out of view).

So testing for left arrow appearing ends up trying to press right again before the animation is done and it breaks.

There's another issue as well as mentioned in my last comment: if the example had more tabs in it, or if we reduce the scroll amount,  there's not always an arrow appearing or disappearing on each press (only the first and last presses). There would be times where both arrows are showing before the press, and continuing to show after press.

The main thing I don't love at the moment is the magical number on the timeout.

Reply:
If we want to go deep on this, we could see if there's a way to hook into/run a callback when the scrolling is finished.

e.g.
https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollend_event

Not supported on Safari tho :(